psi4/src/psi4/libciomr/tstart.cc

```diff
@@ -70,10 +70,10 @@ void PSI_API tstart() {
     const long clk_tck = sysconf(_SC_CLK_TCK);
     times(&total_tmstime);
 
-    // host name has up to HOST_NAME_MAX(==64) + 1 bytes, and must end in the null byte
-    std::vector<char> name(65);
-    error = gethostname(name.data(), 65);
-    if (error != 0) strncpy(name.data(), "nohostname", 11);
+    // host name has up to HOST_NAME_MAX (64 or 256) + 1 bytes, and must end in the null byte
+    std::vector<char> name(257);
+    error = gethostname(name.data(), name.size());
+    if (error) strncpy(name.data(), "nohostname", name.size());
```

Comment:
I wasn't sure about 11 -> name.size() here, but https://cplusplus.com/reference/cstring/strncpy/ convinced me.

overall, lgtm. any concerns, @zachglick ?

Reply:
no concerns, I agree this is a better way to do things. thanks Susi!